src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -214,6 +218,26 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
                 await sessionManager.SetAsync(feature, enabled).ConfigureAwait(false);
             }
 
+            if (featureDefinition.EnableTelemetry)
+            {
+                if (_telemetryPublisher == null)
+                {
+                    string errorMessage = $"The feature declaration enabled telemetry but no telemetry publisher was registered.";
```

Comment:
Looks like we can use the string directly instead of creating the `errorMessage` variable first. Also we don't need the `$` since there's no formatting done inside the string.

Reply:
If there is no formatting done inside the string, we can also use 'const' keyword.